### PR TITLE
Remove deprecated aad pod identity from API docs

### PR DESCRIPTION
**Pull request by dipti-pai**
Remove references to aad pod identity.

### CODE REVIEW

File: docs/spec/v1beta2/kustomizations.md [hunk 1]
```diff
@@ -1260,32 +1260,10 @@ env:
 
 #### Azure Key Vault
 
-While making use of [AAD Pod Identity](https://github.com/Azure/aad-pod-identity),
```
Comment (stefanprodan) [inline docs/spec/v1beta2/kustomizations.md:1263]: Please undo this change, v1beta2 is a deprecated API and its code and docs are considered readonly.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

**Comment by dipti-pai** [comment]
@stefanprodan Thank you for the review :) I have made the requested changes.

**Comment by stefanprodan** [comment]
Please squash all commits into a single one

**Comment by dipti-pai** [comment]
> Please squash all commits into a single one

@stefanprodan Squashed into a single commit. Thanks.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @dipti-pai
